hw/hal/src/hal_flash.c

```diff
@@ -165,6 +167,10 @@ hal_flash_write(uint8_t id, uint32_t address, const void *src,
         return -1;
     }
 
+    if (protected_flash[id / 8] &= (1 << (id & 7))) {
```

Comment:
This looks a bit odd to me.  Did you intend to modify the `protected_flash` array here?  (i.e., should this expression use `&` rather than `&=`?).  This comment applies to a few other uses of `&=` in this file.

Reply:
done.
Thanks @ccollins476ad I must have been brain dead when writing it.